### PR TITLE
Implemented picklist filters and UI for showing recently completed picklists (#1x699zj)

### LINKED ISSUE #82: Show recently completed pick lists
## What is the motivation for adding/enhancing this feature?

Store recently completed pick lists in a state for user reference.

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #82 

### Screenshots of Visual Changes before/after (If There Are Any)

[Screencast from 20-03-23 12:15:09 PM IST.webm](https://user-images.githubusercontent.com/59442907/226266455-3076ceff-cf61-45ec-88a9-8198ea8bec57.webm)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by k2maan** [comment]
Currently, the filter for `only show my picklist` does not work as expected because only the current user's picklists are fetched by API in the first place. This needs to be discussed.

File: src/components/Picklist-item.vue [hunk 1]
```diff
@@ -31,6 +32,9 @@ export default defineComponent({
           this.router.push({ path: `/picklist-details/${picklist.picklistId}` });
         }
       });
+    },
+    getTime(time: any) {
```
Comment (ps011) [inline src/components/Picklist-item.vue:36]: What's the point of using typescript if types are not used?

```suggestion
    getTime(time: number) {
```
https://moment.github.io/luxon/api-docs/index.html#datetimefrommillis

Comment (k2maan) [reply src/components/Picklist-item.vue]: @ps011 Updated, thank you for the review.

File: src/store/modules/picklist/index.ts [hunk 2]
```diff
@@ -9,7 +9,18 @@ const picklistModule: Module<PicklistState, RootState> = {
...
+    picklist: {
+      list: [],
+      total: 0
+    },
+    completed: {
+      list: [],
+      total: 0
+    },
+    filters: {
+      hideCompletedPicklists: false,
```
Comment (adityasharma7) [inline src/store/modules/picklist/index.ts:21]: I think this is not an filter

File: src/store/modules/picklist/actions.ts [hunk 3]
```diff
@@ -10,19 +10,37 @@ const actions: ActionTree<PicklistState, RootState> = {
...
+      },
+      "orderBy": "picklistDate DESC",
+      "viewSize": payload.viewSize,
+      "viewIndex": payload.viewIndex,
+      "fieldList": ["picklistId", "picklistDate", "statusId", "partyId"],
+      "entityName": "PicklistAndRole",
+      "noConditionFind": "Y"
+    } as any
+
+    if (state.filters.showMyPicklists) params.inputFields.partyId = payload.partyId
```
Comment (adityasharma7) [inline src/store/modules/picklist/actions.ts:28]: partyId should be from state

File: src/store/modules/party/actions.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    let resp;
+    const params = {
+      "inputFields": {
+        "partyId": unavailablePickersPartyIds,
+        "userPartyId_op": 'in'
+      },
+      "fieldList": ["firstName", "lastName", "partyId"],
+      "entityName": "PartyAndUserLoginAndPerson",
+      "viewSize": unavailablePickersPartyIds.length,
```
Comment (adityasharma7) [inline src/store/modules/party/actions.ts:23]: Add "distinct": "Y"

File: src/store/modules/picklist/actions.ts [hunk 5]
```diff
@@ -31,6 +49,46 @@ const actions: ActionTree<PicklistState, RootState> = {
...
+        let list = resp.data.docs;
+        const pickersPartyIds = [...new Set(list.map((item: any) => item.partyId))]
+        const pickersDetails = await dispatch('party/getPickersDetails', pickersPartyIds, { root: true });
+
+        list = list.map((item: any) => ({ ...item, pickersFullName: pickersDetails[item.partyId].fullName }))
+
+        commit(types.PICKLISTS_COMPLETED_UPDATED, { list: list, total: resp.data.count });
+        return resp.data;
+      } else {
+        commit(types.PICKLISTS_COMPLETED_UPDATED, { list: [], total: 0 });
```
Comment (adityasharma7) [inline src/store/modules/picklist/actions.ts:83]: We should initialise default values, and have single commit at last with response values. If it fails, initial value will be set at last

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**